### ISSUE TITLE
Consider self-contained doctests

### ISSUE THREAD

**Issue by dmitry-timofeev** [jakewins/javadoctest#1]
Hi, @jakewins,

Thank you for sharing this project! I have been looking for something similar to Rust [documentation tests](https://doc.rust-lang.org/rustdoc/documentation-tests.html) but for Java after working with a Rust project. Your tool appeared very similar. However, in Rust the code snippets are supposed to be self-contained tests (i.e., they don’t require a separate test class, hence, no need to reference it). I built a prototype that brings such approach to `javadoctest` (will push shortly).

  For example, it will discover and execute a snippet with class «doctest», no extra tags needed:

```
<pre class="doctest">
  assert "Hello".length() == 10; // Will fail
</pre>
```

It is based on JUnit Platform framework, which enables some features:

 - integration with existing test plugins supporting JUnit Platform (e.g., [junit-platform-maven-plugin](https://github.com/sormuras/junit-platform-maven-plugin), or [Gradle](https://junit.org/junit5/docs/current/user-guide/#running-tests-build-gradle)), bringing easy configuration
 - test reporting in JUnit format for free

Would you be interested to go with such approach for 2.x? Or is it too much of a change from the current approach?

**Comment by jakewins** [comment]
Hey, sorry for the slow turn-around here; yes, this seems very cool. I'll merge your PR.

**Comment by jakewins** [comment]
Oh; @dmitry-timofeev do you mind updating the README?

**Comment by dmitry-timofeev** [comment]
Hi @jakewins , no problem, thank you for reviewing! If you are fine with the approach, shall I move the remaining tasks for various enhancements from a private Trello board to this project tracker?

> do you mind updating the README?

Not at all, however, it's a little too early: there is an issue that I discovered with the launching the engine through Maven — it is also needed to pass the project classpath for correct compilation/execution. I've documented that issue in #3. I am not able to allocate enough time for it now, but I hope I'll get to it.

Would it make sense then to mark the current HEAD as `develop`, and reset `master` back to the previous version, so that the new users don't get confused?
